File: taller/SergioDeIscar-taller-java/SergioDeIscar-taller-java/src/main/java/org/example/models/Trabajador.java
package org.example.models;

public abstract class Trabajador extends Persona{
    public Trabajador(String nombre, int experiencia, int horasDiarias){
        super(nombre, experiencia);
        setHorasDiarias(horasDiarias);
    }

    int salario = 1800;

    protected int horasDiarias;

    @Override
    void saludar() {
        System.out.println("Trabajador " + nombre + " saluda con respeto.");
    }

    public void descansar(){
        System.out.println("El trabajador " + nombre + " descansa, con un horario de " + horasDiarias + " horas.");
    }

    abstract void comer();

    //region Getter
    int getHorasDiarias(){
        return horasDiarias;
    }
    //endregion

    //region Setter
    private void setHorasDiarias(int horasDiarias){
        if (horasDiarias < 0 || horasDiarias > 13) return;
        this.horasDiarias = horasDiarias;
    }
    //endregion
}

File: taller/SergioDeIScar-taller-kotlin/SergioDeIscar-taller-kotlin/src/main/kotlin/models/Trabajador.kt
package taller.models

abstract class Trabajador(nombre: String, experiencia: Int, val horasDiarias: Int): Persona(nombre, experiencia) {
    fun descansar(){
        println("El trabajador $nombre descansa, con un horario de $horasDiarias horas.")
    }
    override fun saludar() {
        println("Trabajador $nombre saluda con respeto.")
    }
    abstract fun comer()

    override fun equals(other: Any?): Boolean {
        if (other !is Trabajador) return false
        val trabajador = other as Trabajador
        if (trabajador === this) return true
        return trabajador.nombre == this.nombre &&
                trabajador.experiencia == this.experiencia &&
                trabajador.horasDiarias == this.horasDiarias &&
                trabajador.salario == this.salario
    }
}